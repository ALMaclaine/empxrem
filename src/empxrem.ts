export function pxToRem(px: number, baseFontSize: number = 16) {
    return `${px / baseFontSize}rem`;
}

export function pxToEm(px: number, baseFontSize: number = 16) {
    return `${px / baseFontSize}em`;
}

export function ensureNotPx(unit: number | string) {
    if (typeof unit === 'string') {
        if (unit.includes('px')) {
            const val = parseFloat(unit.replace('px', ''));
            unit = pxToRem(val);
        }
    } else {
        unit = pxToRem(unit);
    }
    return unit;
}

export function ensureNotPxEm(unit: number | string) {
    if (typeof unit === 'string') {
        if (unit.includes('px')) {
            const val = parseFloat(unit.replace('px', ''));
            unit = pxToRem(val);
        }
    } else {
        unit = pxToEm(unit);
    }
    return unit;
}
