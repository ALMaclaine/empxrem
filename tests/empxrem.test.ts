import * as npmUtils from '../src';

describe('empxrem', () => {
  it('Has Correct API', () => {
    const keys = Object.keys(npmUtils);
    expect(keys.length).toBe(4);
    expect(keys).toMatchSnapshot();
  });
});
