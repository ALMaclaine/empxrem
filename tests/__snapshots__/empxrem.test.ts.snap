// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empxrem Has Correct API 1`] = `
Array [
  "pxToRem",
  "pxToEm",
  "ensureNotPx",
  "ensureNotPxEm",
]
`;
